packages/api/endpoints/granules.js

```diff
@@ -569,20 +580,32 @@ async function bulkOperations(req, res) {
   return res.status(202).send(asyncOperation);
 }
 
+const BulkDeletePayloadSchema = z.object({
+  forceRemoveFromCmr: z.boolean().optional(),
+  concurrency: z.number().int().positive().optional(),
+  maxDbConnections: z.number().int().positive().optional(),
+  knexDebug: z.boolean().optional(),
+}).catchall(z.unknown());
+
+const parseBulkDeletePayload = zodParser('Bulk delete payload', BulkDeletePayloadSchema);
+
 /**
  * Start an AsyncOperation that will perform a bulk granules delete
  *
- * @param {Object} req - express request object
- * @param {Object} res - express response object
- * @returns {Promise<Object>} the promise of express response object
+ * @param {Request} req - express request object
+ * @param {Response} res - express response object
+ * @returns {Promise<unknown>} the promise of express response object
  */
 async function bulkDelete(req, res) {
-  const payload = req.body;
-
-  if (payload.forceRemoveFromCmr && !isBoolean(payload.forceRemoveFromCmr)) {
-    return res.boom.badRequest('forceRemoveFromCmr must be a boolean value');
+  const payload = parseBulkDeletePayload(req.body);
+  if (isError(payload)) {
+    return res.status(400).send({ errors: payload.errors });
```

Comment:
This should be noted as a breaking change since the API will be returning different error output now. Still worth doing, though.

Reply:
Hmm, that's a good point, I had been thinking it returns the same status code, but the *text* return is changing. 

In that case..... I think I'm going to remove zod from this PR and reset the responses, as we don't want to be making breaking changes to the backport release series unless it's a critical bugfix.

Reply:
On second thought -- we could maintain error output via this sort of thing and avoid 'breaking' changes for the existing error check: 

2812e1f9d1

Opinions?

Reply:
On second thought -- we could maintain error output via this sort of thing and avoid 'breaking' changes for the existing error check:

2812e1f9d1

Opinions?